samples/algorithms/CatState.qs

```diff
@@ -21,20 +22,19 @@ namespace Sample {
         DumpMachine();
 
         // Measure and reset qubits before releasing them.
-        let results = MeasureEachZ(register);
-        ResetAll(register);
-        return results;
+        MResetEachZ(register)
     }
 
-    // Prepares a cat state 1/sqrt(2)(|0...0〉 + |1...1〉) using a qubit register
-    // in the zero state.
+    /// # Summary
+    /// Prepares state (|0...0〉 + |1...1〉) / √2 (a generalized GHZ state
+    /// or a cat state) across the `register` of qubits.
+    /// All qubits are assumed to be in |0〉 state on input.
     operation PrepareCatState(register : Qubit[]) : Unit {
         Fact(Length(register) > 0, "Qubit register must not be empty.");
-        Fact(CheckAllZero(register), "Qubits are not in the |0〉 state.");
 
-        // Set the first qubit in the register into a 1/sqrt(2)(|0〉 + |1〉) superposition.
+        // Set the first qubit in the register into a (|0〉 + |1〉) / √2 superposition.
         // Then apply a CNOT to the remaining qubits using the first qubit as control.
         H(register[0]);
-        ApplyToEach(CNOT(register[0], _), register[1...]);
+        ApplyToEach(CNOT(register[0], _), Rest(register));
```

Comment:
What was wrong with the `[1...]` syntax? That seems to work fine for generating QIR and avoids the need to open the arrays namespace.

Reply:
Yes, I like [1...] and I changed the code back.
Funny, just a few months ago I was advocating the same thing - if we have a concise way of expressing something in a language we should prefer it over the library function. I'm still advocating it now. It's just I was looking at library code, library documentation and the set of function a lot recently that it got into my brain. :)